langchain/document_loaders/directory.py

```diff
@@ -64,27 +81,22 @@ def load(self) -> List[Document]:
             except ImportError as e:
                 logger.warning(
                     "To log the progress of DirectoryLoader you need to install tqdm, "
-                    "`pip install tqdm`"
-                )
+                    "`pip install tqdm`")
                 if self.silent_errors:
                     logger.warning(e)
                 else:
                     raise e
 
-        for i in items:
-            if i.is_file():
-                if _is_visible(i.relative_to(p)) or self.load_hidden:
-                    try:
-                        sub_docs = self.loader_cls(str(i), **self.loader_kwargs).load()
-                        docs.extend(sub_docs)
-                    except Exception as e:
-                        if self.silent_errors:
-                            logger.warning(e)
-                        else:
-                            raise e
-                    finally:
-                        if pbar:
-                            pbar.update(1)
+        if self.parallel:
+            results = process_map(self._load_file, items) # use default max_workers
```

Comment:
where is process_map defined?

Reply:
it's from [tqdm](https://tqdm.github.io/docs/contrib.concurrent/)

The reason I used it is to support the progress bar for parallel loading. 

When I have some time I will update the PR to only use tqdm library with parallel or progress bar and use the original implementation otherwise to keep tqdm optional

Reply:
don't we need to import it somewhere? am i just not seeing it?

Reply:
Yup the import line is missing I  don't know how this happened since I committed all changes